### ISSUE TITLE
[BUG] Vision.__init__ hardcodes self.dim = 2048 regardless of backbone

### ISSUE THREAD

**Issue by sachncs** [sachncs/morel#10]
## Describe the Bug

`morel.data.extract.Vision` sets `self.dim = 2048` unconditionally for every torchvision backbone. The class docstring says "Output feature dimension" as if it derives from the model, but it does not — it assumes ResNet50.

```python
# morel/data/extract.py:175-196
class Vision:
    def __init__(self, model: str, *, batch: int = 32) -> None:
        builder = getattr(tv_models, model, None)
        if builder is None:
            raise Datum(f"torchvision has no model named {model!r}")
        self.name = model
        self.batch = batch
        self.model = builder(weights="DEFAULT")
        self.model.fc = torch.nn.Identity()
        self.model.eval()
        self.dim = 2048  # <-- hardcoded; ignores builder(...).fc.in_features
```

A user who passes `model="resnet18"` gets a 512-dim feature but reports 2048 to the pipeline, silently corrupting every downstream shape.

## To Reproduce

1. `from morel.data.extract import Vision`
2. `v = Vision("resnet18")`
3. `import torchvision.models as tvm; assert v.dim == tvm.resnet18(weights=None).fc.in_features` → fails (expected 512, got 2048).

## Expected Behavior

`self.dim = builder(weights=None).fc.in_features` (or equivalent) so the reported dimension matches the actual backbone output.

## Environment

- OS: any
- Python version: ≥3.10
- Package version: 0.1.dev76+

## Additional Context

Engineering audit finding `C-18` (High). Silent shape mismatch is a real correctness hazard. `# pragma: no cover` on `encode` means this is currently untested.

## Checklist

- [x] I have searched the existing issues.
- [x] I have provided a minimal reproducible example.
- [x] I have included the environment details.

**Comment by sachncs** [comment]
Resolved in master. The relevant fix is split across one or two commits; see the change history since the issue was opened.
